### PR TITLE
fix(#49): show error when navigating to non-existing ticket url

### LINKED ISSUE #49: Show error when navigating to non-existing ticket url
The app currently crashes when navigating to a non-existent ticket url. We should handle this by letting the user know the current url is invalid, and give the user the chance to navigate home and submit a new ticket, or select an existing ticket from the  RecentTickets component

### PR DESCRIPTION
**Pull request by ThomasRiddle**
Fix #49 .
In this PR:
- Adding the page if the ticket does not exist in views/tickets/TicketNotFound.js.
- Navigating to the page if the ticket does not exist. 
Screen Shot 2020-02-07 at 14 26 53

Announcements:
- Try to keep the same page format as PR #63 .
- Error handling format follows [React Error Boundaries](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html).

Question(s):
- The error code is currently hardcoded, do we need to get them from the error messages?
- Should I follow the format that we always used?

### CODE REVIEW

File: src/views/tickets/Ticket.js [hunk 1]
```diff
@@ -50,51 +52,60 @@ export default class Ticket extends Component {
...
       // todo: display the error on the page
+      this.setState({ hasError: true });
     }
   };
 
   onFileSelectionChange = (e, data) => {
     this.setState({ currentCode: data.value });
   };
 
+  static getDerivedStateFromError = error => {
```
Comment (williamtdr) [inline src/views/tickets/Ticket.js]: Is this used?

File: src/views/tickets/Ticket.js [hunk 2]
```diff
@@ -50,51 +52,60 @@ export default class Ticket extends Component {
       }
     } catch (error) {
       // todo: display the error on the page
```
Comment (williamtdr) [inline src/views/tickets/Ticket.js]: Can you take out this comment now that the error is being shown?

File: src/views/tickets/Ticket.js (hunk 1 above)
Comment (ThomasRiddle) [reply src/views/tickets/Ticket.js]: It's helpful, huge thanks!

File: src/views/tickets/Ticket.js (hunk 2 above)
Comment (ThomasRiddle) [reply src/views/tickets/Ticket.js]: Sure, I forget to do this.

**Comment by ThomasRiddle** [comment]
Now looks like this.
Screen Shot 2020-02-07 at 15 02 45

**Comment by williamtdr** [comment]
`getDerivedStateFromError` is still there, but as far as I can tell isn't being used by anything... does it still work without it?

**Comment by ThomasRiddle** [comment]
> `getDerivedStateFromError` is still there, but as far as I can tell isn't being used by anything... does it still work without it?

It actually works. However, I have read the [documentation here](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html) and they use it. I am not sure it's safe for us to remove the function.

**Comment by williamtdr** [comment]
Makes sense, React paradigms are always changing... LGTM.

**Pull request merged**